autoemulate/emulators/gaussian_process_torch.py

```diff
@@ -189,23 +209,90 @@ def predict(self, X, return_std=False):
             return mean, std
         return mean
 
-    def get_grid_params(self, search_type="random"):
+    def get_grid_params(self, search_type="random", input_dim=1):
         """Returns the grid parameters for the emulator."""
-        param_space = {
-            "covar_module": [
-                # TODO: initialize lengthscale for other kernels?
-                gpytorch.kernels.RBFKernel().initialize(lengthscale=1.0),
-                gpytorch.kernels.MaternKernel(nu=2.5),
-                gpytorch.kernels.MaternKernel(nu=1.5),
-                gpytorch.kernels.PeriodicKernel(),
-                gpytorch.kernels.RQKernel(),
-            ],
-            "mean_module": [gpytorch.means.ConstantMean(), gpytorch.means.ZeroMean()],
-            "optimizer": [torch.optim.AdamW, torch.optim.Adam, torch.optim.SGD],
-            "lr": [5e-1, 1e-1, 5e-2, 1e-2],
-            "max_epochs": [50, 100, 200],
-            "normalize_y": [True, False],
-        }
+        if search_type == "random":
+            param_space = {
+                "covar_module": [
+                    # TODO: initialize lengthscale for other kernels?
+                    gpytorch.kernels.RBFKernel(ard_num_dims=input_dim).initialize(
+                        lengthscale=1.0
+                    ),
+                    gpytorch.kernels.MaternKernel(nu=2.5, ard_num_dims=input_dim),
+                    gpytorch.kernels.MaternKernel(nu=1.5, ard_num_dims=input_dim),
+                    gpytorch.kernels.PeriodicKernel(),
+                    gpytorch.kernels.RQKernel(ard_num_dims=input_dim),
+                    gpytorch.kernels.ScaleKernel(
+                        gpytorch.kernels.MaternKernel(ard_num_dims=input_dim)
+                    )
```

Comment:
so the gp here is a multitask gp, and in the [docs](https://docs.gpytorch.ai/en/stable/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html) they say 

> You may also notice that we don’t use a ScaleKernel, since the MultitaskKernel will do some scaling for us. (This way we’re not overparameterizing the kernel.)

So I think this can be taken out for now.  However, I'm just thinking about how to best have multiple gpytorch models, and then we'll have a gp with independent output dimensions where we'll wrap the kernels into ScaleKernels.

Also, on a side-note, I'd be curious about your experience with combined kernels (multiplicative/additive) and whether there's anything else we should add.

Reply:
I wasn't aware of that `MultitaskKernel` was doing some scaling.

I have a fairly limited understanding of the advantage of using combined kernels. From what I have seen there isn't any clear advantage, but I might be proven wrong.